sqlx-core/src/sqlite/statement/worker.rs

```diff
@@ -31,14 +32,19 @@ impl StatementWorker {
             for cmd in rx {
                 match cmd {
                     StatementWorkerCommand::Step { statement, tx } => {
-                        let status = unsafe { sqlite3_step(statement.0.as_ptr()) };
+                        let resp = if let Some(statement) = statement.upgrade() {
+                            let status = unsafe { sqlite3_step(statement.0.as_ptr()) };
 
-                        let resp = match status {
-                            SQLITE_ROW => Ok(Either::Right(())),
-                            SQLITE_DONE => Ok(Either::Left(statement.changes())),
-                            _ => Err(statement.last_error().into()),
+                            let resp = match status {
+                                SQLITE_ROW => Ok(Either::Right(())),
+                                SQLITE_DONE => Ok(Either::Left(statement.changes())),
+                                _ => Err(statement.last_error().into()),
+                            };
+                            resp
+                        } else {
+                            // Statement is already finalized.
+                            Err(Error::WorkerCrashed)
```

Comment:
Strictly speaking, the worker didn't actually crash here. I'd consider a different error variant, maybe even a new one if none of the existing fit.